Automattic/wp-calypso
# Improve domain actions permissions with new api properties

## Changes proposed in this Pull Request

This PR aims to add some additional checks to edit domain contact informations, dns records and name servers, using new domain property added in this diff D76803-code

![dispute-flag-settings](https://user-images.githubusercontent.com/2797601/158556496-f0b9b6d5-2eab-4135-b19d-ff3161bfe0b2.png)

![dispute-flag-dns](https://user-images.githubusercontent.com/2797601/158556519-3b186e10-2fa9-4407-b92f-45cf393b1d01.png)

<img width="1215" alt="Schermata 2022-03-16 alle 10 23 36" src="https://user-images.githubusercontent.com/2797601/158558276-630d0823-5dcd-47a4-a47c-8bd1548d2904.png">

## Testing instructions

- Apply the back-end patch D76803-code
- Build this branch locally or open the live Calypso link
- Set a dispute flag on a test domain
- Open domain settings page for that domain and verify that _Name servers_, _DNS records_ and _Contact Information_ cards show a warning message
- Verify that _DNS page_ (`domains/manage/all/{:domain}/dns/{:site}`) shows a warning message
- Verify that _Edit Contact Info page_ (`domains/manage/all/{:domain}/edit-contact-info/{:site}`) shows a warning message
